youtube_dl/extractor/imgur.py

```diff
@@ -102,23 +102,32 @@ def _real_extract(self, url):
 class ImgurAlbumIE(InfoExtractor):
     _VALID_URL = r'https?://(?:i\.)?imgur\.com/gallery/(?P<id>[a-zA-Z0-9]+)'
 
-    _TEST = {
+    _TESTS = [{
         'url': 'http://imgur.com/gallery/Q95ko',
         'info_dict': {
             'id': 'Q95ko',
         },
         'playlist_count': 25,
-    }
+    },{
+        'url': 'https://imgur.com/gallery/YcAQlkx',
+        'info_dict': {
+            'id': 'YcAQlkx',
+        },
+        'playlist_count': 1,
+    }]
 
     def _real_extract(self, url):
         album_id = self._match_id(url)
 
-        album_images = self._download_json(
-            'http://imgur.com/gallery/%s/album_images/hit.json?all=true' % album_id,
-            album_id)['data']['images']
+        album_img_data = self._download_json(
+            'http://imgur.com/gallery/%s/album_images/hit.json?all=true' % album_id,album_id)['data']
 
-        entries = [
-            self.url_result('http://imgur.com/%s' % image['hash'])
-            for image in album_images if image.get('hash')]
+        if(len(album_img_data) == 0):
+            entries = [self.url_result('http://imgur.com/%s' % album_id)]
```

Comment:
"Just return directly"
Do you mean something like this:
```
if len(album_img_data) == 0:
    return {
    	'id': album_id
    	'url': 'http://imgur.com/%s' % album_id
    	'title': self._og_search_title(
    		self._download_webpage(
    			compat_urlparse.urljoin(url, album_id), album_id)
    		)	#title is required (without it complains about missing title)
    	
    }
```
Apparently that too produces some error:
```
>  python test/test_download.py TestDownload.test_ImgurAlbum_1
E
======================================================================
ERROR: test_ImgurAlbum_1 (__main__.TestDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_download.py", line 93, in test_template
    raise Exception('Test definition incorrect. The output file cannot be known. Are both \'id\' and \'ext\' keys present?')
Exception: Test definition incorrect. The output file cannot be known. Are both 'id' and 'ext' keys present?

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
```